modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_2/TemplateUpgradeProcess.java

```diff
@@ -23,28 +24,37 @@
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 
+import java.util.ArrayList;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /**
  * @author Albert Gomes Cabral
+ * @author Felipe Veloso
  */
-public abstract class BaseTemplateUpgradeProcess extends UpgradeProcess {
+public class TemplateUpgradeProcess extends UpgradeProcess {
 
 	@Override
 	protected void doUpgrade() throws Exception {
-		_upgradeDDMTemplates();
-		_upgradeFragmentEntries();
-	}
-
-	protected String getTemplateContextVariable() {
-		return null;
+		ArrayList<Template> templates = new ArrayList<>();
+
+		templates.add(
+			new Template(
+				"\\w+\\s*\\=\\s*.+com\\.liferay\\.portal\\.kernel\\." +
+					"servlet\\.BrowserSnifferUtil\\\"\\)",
+				StringPool.BLANK, "browserSniffer"));
+
+		templates.add(
+			new Template(
+				"\\w+\\s*=\\s*serviceContext\\.getThemeDisplay\\(\\)",
+				StringPool.BLANK, "themeDisplay"));
+
+		for (Template template : templates) {
+			_upgradeDDMTemplates(template);
+			_upgradeFragmentEntries(template);
```

Comment:
@velosofelipe64, this way we're still rerunning the SQL queries, therefore in order to run them a single time, we should use this `for` inside the methods `_upgradeDDMTemplates` and `_upgradeFragmentEntries`. You may try to make the variable `templates` an attribute of this class, populate it with the templates information and get it inside the methods `_upgradeDDMTemplates` and `_upgradeFragmentEntries`.

Does it make sense to you?

Reply:
Yes, sure.
Do you think we can simplify the methods `_upgradeDDMTemplates` and `_upgradeFragmentEntries` in only one, with the variable `templates` and `selectPreparedStatement` as parameters?